src/include/migraphx/op/roialign.hpp

```diff
@@ -0,0 +1,321 @@
+        }
+
+        std::vector<std::size_t> out_lens = x_lens;
+        out_lens[0]                       = roi_lens[0];
+        out_lens[2]                       = output_height;
+        out_lens[3]                       = output_width;
+
+        return {type, out_lens};
+    }
+
+    struct pos_weight
+    {
+        int pos1;
+        int pos2;
+        int pos3;
+        int pos4;
+        float w1;
+        float w2;
+        float w3;
+        float w4;
+    };
+
+    void calc_pos_weight(const int64_t height,
+                         const int64_t width,
+                         const shape& comp_s,
+                         float roi_start_h,
+                         float roi_start_w,
+                         float bin_size_h,
+                         float bin_size_w,
+                         int64_t roi_bin_grid_h,
+                         int64_t roi_bin_grid_w,
+                         std::vector<pos_weight>& pos_weights) const
+    {
+        shape_for_each(comp_s, [&](auto idx) {
+            auto ph    = idx[0];
+            auto pw    = idx[1];
+            auto iy    = idx[2];
+            auto ix    = idx[3];
+            auto index = comp_s.index(idx);
+            const float yy =
+                roi_start_h + ph * bin_size_h + (iy + .5f) * bin_size_h / roi_bin_grid_h;
+            const float xx =
+                roi_start_w + pw * bin_size_w + (ix + .5f) * bin_size_w / roi_bin_grid_w;
+
+            float x = (coord_trans_mode == "output_half_pixel") ? (xx - 0.5f) : xx;
+            float y = (coord_trans_mode == "output_half_pixel") ? (yy - 0.5f) : yy;
+
+            // deal with: inverse elements are out of feature map boundary
+            if(y < -1.0 || y > height || x < -1.0 || x > width)
+            {
+                auto& pc = pos_weights[index];
+                pc.pos1  = 0;
+                pc.pos2  = 0;
+                pc.pos3  = 0;
+                pc.pos4  = 0;
+                pc.w1    = 0;
+                pc.w2    = 0;
+                pc.w3    = 0;
+                pc.w4    = 0;
+                return;
+            }
+
+            y          = (y <= 0) ? 0 : y;
+            x          = (x <= 0) ? 0 : x;
+            auto y_low = static_cast<int64_t>(y);
+            auto x_low = static_cast<int64_t>(x);
+            int64_t y_high;
+            int64_t x_high;
+
+            y_high = y_low + 1;
+            if(y_low >= height - 1)
+            {
+                y = y_high = y_low = height - 1;
+            }
+
+            x_high = x_low + 1;
+            if(x_low >= width - 1)
+            {
+                x = x_high = x_low = width - 1;
+            }
+
+            float ly = y - y_low;
+            float lx = x - x_low;
+            float hy = 1.0f - ly;
+            float hx = 1.0f - lx;
+            float w1 = hy * hx;
+            float w2 = hy * lx;
+            float w3 = ly * hx;
+            float w4 = ly * lx;
```

Comment:
Can you just assign the values directly to `pos_weight`?

Reply:
fixed